### PR TITLE
Fixed the default behaviour of the autopartitioning (#1380767)

### PR DESCRIPTION
**Pull request by poncovka**
Always set the default partitioning
* The default partitioning of the install class should be always set.

Enable to define the autopart type in an install class
* The default partitioning scheme can be defined in an install class.
  If the kickstart file defines the scheme, it will be used instead.
    
Resolves: rhbz#1380767

### CODE REVIEW

**Comment by Frodox** [comment]
Great +1 for this!

File: pyanaconda/kickstart.py [hunk 1]
```diff
@@ -290,7 +290,7 @@ def execute(self, storage, ksdata, instClass):
 
         # sets up default autopartitioning.  use clearpart separately
         # if you want it
-        instClass.setDefaultPartitioning(storage)
+        refreshAutoSwapSize(storage)
```
Comment (vpodzime) [inline pyanaconda/kickstart.py]: Please also describe this particular change in the commit message.

File: pyanaconda/installclass.py [hunk 2]
```diff
@@ -58,6 +58,10 @@ class BaseInstallClass(object):
     # Blivet uses by default.
     defaultFS = None
 
+    # The default partitioning scheme to use for autopartitioning.
+    # If None, we will use DEFAULT_AUTOPART_TYPE.
```
Comment (vpodzime) [inline pyanaconda/installclass.py]: Why is it being set to ``None`` then?

File: pyanaconda/iutil.py [hunk 3]
```diff
@@ -1242,3 +1242,10 @@ def touch(file_path):
     # even when the path points to dirrectory
     if not os.path.exists(file_path):
         os.mknod(file_path)
+
+def firstNotNone(*args):
+    """Return the first argument that is not None."""
+    for arg in args:
+        if arg is not None:
+            return arg
```
Comment (vpodzime) [inline pyanaconda/iutil.py]: This whole function can be written as:
```
return next((arg for arg in args if arg is not None), None)
```

Comment (vpodzime) [inline pyanaconda/iutil.py]: And if it is a function, it should be generic -- i.e. take any iterable as a single argument and return the first item that is not None. AFAICT there's no need to use ``*args`` machinery here.

File: pyanaconda/ui/tui/spokes/storage.py [hunk 4]
```diff
@@ -42,15 +42,15 @@
...
+from pykickstart.constants import CLEARPART_TYPE_ALL, CLEARPART_TYPE_LINUX, CLEARPART_TYPE_NONE
 from pykickstart.errors import KickstartValueError
 
 from collections import OrderedDict
 
 import logging
 log = logging.getLogger("anaconda")
 
-__all__ = ["StorageSpoke", "AutoPartSpoke"]
+__all__ = ["StorageSpoke"]
```
Comment (vpodzime) [inline pyanaconda/ui/tui/spokes/storage.py]: Is this change intentional? What is the purpose?

File: pyanaconda/installclass.py (hunk 2 above)
Comment (poncovka) [inline pyanaconda/installclass.py]: I could change `None` to `DEFAULT_AUTOPART_TYPE`, but then it is not necessary to have the constant `DEFAULT_AUTOPART_TYPE` at all, because it will be used only here.

File: pyanaconda/iutil.py (hunk 3 above)
Comment (poncovka) [inline pyanaconda/iutil.py]: I've seen this implementation before, but it is really hard to read. What are the advantages?

Comment (poncovka) [inline pyanaconda/iutil.py]: I can make the function generic, though,

File: pyanaconda/ui/tui/spokes/storage.py (hunk 4 above)
Comment (poncovka) [inline pyanaconda/ui/tui/spokes/storage.py]: In tui, it does not make sense to enable automatic import of indirect spokes, because we always create them and use them only in other spokes. With automatic import, they will be found by a hub, created, initialized and forgotten.

Also, I've added a new parameter to the `AutoPartSpoke.__init__` method and the initialization in a hub does not work anymore.

File: pyanaconda/kickstart.py (hunk 1 above)
Comment (poncovka) [inline pyanaconda/kickstart.py]: Fixed.

File: pyanaconda/installclass.py (hunk 2 above)
Comment (vpodzime) [inline pyanaconda/installclass.py]: Still better to have a constant for this -- for other installclasses, addons,...

File: pyanaconda/iutil.py (hunk 3 above)
Comment (vpodzime) [inline pyanaconda/iutil.py]: There's no advantage, it just looks more "pythonic", whatever that means. :) Feel free to leave the for cycle in place. My main concern here is the (non-)genericity.

File: pyanaconda/ui/tui/spokes/storage.py (hunk 4 above)
Comment (vpodzime) [inline pyanaconda/ui/tui/spokes/storage.py]: Thanks for the extra explanation. I think this deserves a separate commit or at least some explanation in the commit message, though.

Comment (poncovka) [inline pyanaconda/ui/tui/spokes/storage.py]: Fixed.

File: pyanaconda/installclass.py (hunk 2 above)
Comment (poncovka) [inline pyanaconda/installclass.py]: Fixed.

File: pyanaconda/iutil.py (hunk 3 above)
Comment (poncovka) [inline pyanaconda/iutil.py]: Fixed.
